pkg/istio-agent/xds_proxy.go

```diff
@@ -84,6 +87,15 @@ type XdsProxy struct {
 	connected      *ProxyConnection
 	initialRequest *discovery.DiscoveryRequest
 	connectedMutex sync.RWMutex
+
+	// Wasm cache and ecds channel are used to repalce wasm remote load with local file.
+	wasmCache      wasm.Cache
+	ecdsUpdateChan chan *discovery.DiscoveryResponse
+	// ecds version and nonce uses atomic only to prevent race in testing.
+	// In reality there should not be race as istiod will only have one
+	// in flight update for each type of resource.
+	ecdsLastAckVersion atomic.String
```

Comment:
Each resource in ECDS is potentially independently versioned. Is this assuming something about istiod that is not part of xDS?

Reply:
Yeah, it relies on the fact that istiod version ecds the same in an update. I added a todo for this to better accommodate different behavior or ecds backends. Let me know if this is ok.

Reply:
That's fine since the flag can be used to turn it off for non-istiod xDS server

Reply:
How can someone have different versions? Is it only for non-ADS?